cpu/sam0_common/periph/rtc.c

```diff
@@ -94,21 +133,36 @@ void rtc_init(void)
     OSC32KCTRL->RTCCTRL.reg = OSC32KCTRL_RTCCTRL_RTCSEL_ULP1K;
 
 #endif /* ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE */
+}
+#endif /* CPU_SAMD21 - Clock Setup */
 
-    /* Software Reset the RTC */
-    RTC->MODE2.CTRLA.bit.SWRST = 1;
-    /* Wait end of reset */
-    while (RTC->MODE2.CTRLA.bit.SWRST);
+void rtc_init(void)
+{
+    /* DISABLE RTC MASTER */
+    rtc_poweroff();
+
+    rtc_clock_setup();
```

Comment:
Please call `rtc_clock_setup();` before `rtc_poweroff();`
SAML10-XPRO get stuck within `rtc_poweroff();` which is a normal behavior since we're waiting for a bit but the IP is not clocked.
Strangely, it works so far for SAMD21, SAML21 and SAML11. Atmel black magic ?

Reply:
Thank you for testing!
btw, do we need to call `rtc_poweroff()` and `rtc_reset()` here at all?

e.g. the RTC keeps ticking in HIBERNATE / BACKUP deep sleep mode and can wake the CPU (which triggers a reset).

I was wondering why the RTC was still starting at 0 after that - only to discover that it is being reset in `rtc_init()`! 

Is this the intended behavior? If not, I'd much rather just remove those calls here.

Reply:
@benpicco I have no strong opinion about `reset` yet. But rtc_poweroff() is mandatary for at least SAML21.
CTRLA register at least is "Enable-Protected" which means we cannot modify it after the ENABLE bit has beent set. This is why we must use `rtc_poweroff();` before configuring our rtc.
Try to run the current code on SAML21, it should hang.

Reply:
Ok, I've changed the call order and made the RTC persist across reboots.
I might as well remove `rtc_reset()` then as it's currently unused.

Only tested on the samr21 as I don't have a saml21.